src/lib/common/include/sol-certificate.h

```diff
@@ -65,14 +65,18 @@ struct sol_cert *sol_cert_load_from_file(const char *filename);
  */
 void sol_cert_unref(struct sol_cert *cert);
 
+#ifdef FEATURE_FILESYSTEM
```

Comment:
noooo... see there is no `SOL_` prefix in that symbol, this is a hint this symbol is not exported in our headers. You need to see in the generated `sol-common-buildopts.h` (from `sol-common-buildopts.h.in`) if there is a symbol, if not add it.